src/main/java/org/apache/lucene/util/ReusableStringReaderCloner.java

```diff
@@ -35,16 +34,18 @@
  * private field {@code String s}, storing the original content.
  * It is therefore sensitive to Lucene implementation changes.
  */
-public class ReusableStringReaderCloner implements ReaderCloneFactory.ReaderCloner<ReusableStringReader> {
+public class ReusableStringReaderCloner implements ReaderCloneFactory.ReaderCloner<Reader> {
 
     private static java.lang.reflect.Field internalField;
+    private static Class<Reader> reusableStringReader;
 
-    private ReusableStringReader original;
+    private Reader original;
     private String originalContent;
 
     static {
         try {
-            internalField = ReusableStringReader.class.getDeclaredField("s");
+            reusableStringReader = (Class<Reader>) ReusableStringReaderCloner.class.getClassLoader().loadClass("org.apache.lucene.analysis.ReusableStringReader");
```

Comment:
This is a glorious glorious hack.

In ES 2.0, plugins get a separate class loader. Package equality are dependent on BOTH the package name and the classloader that loaded the package. Thus, the package name of this class is no longer equal to the org.apache.lucene.analysis package when loaded in ES (I also moved it to clarify). Instead, we manually load the actual class.